### PR TITLE
Propagate parameter changes to decompress child nodes

### PR DESCRIPTION
**Pull request by jnidzwetzki**
This patch ensures that changed parameters are propagated to the child nodes of the DecompressChunk node before a rescan is executed.

Fixes: #6069

### CODE REVIEW

File: tsl/test/sql/compression.sql [hunk 1]
```diff
@@ -915,3 +915,86 @@ INSERT INTO ht_metrics_partially_compressed VALUES ('2020-01-01'::timestamptz, 1
...
+
+-- Test parameter change on rescan
+-- issue 6069
+CREATE TABLE IF NOT EXISTS i6069 (
+	timestamp TIMESTAMP WITHOUT TIME ZONE NOT NULL,
+	attr_id SMALLINT NOT NULL,
+	number_val DOUBLE PRECISION DEFAULT NULL
+);
+
+SELECT create_hypertable(
```
Comment (svenklemm) [inline tsl/test/sql/compression.sql:927]:
```suggestion
SELECT table_name FROM create_hypertable(
```

**Review by svenklemm — APPROVED** [review]

File: tsl/test/sql/compression.sql (hunk 1 above)
Comment (jnidzwetzki) [reply tsl/test/sql/compression.sql]: Changed

File: .unreleased/bugfix_6084 [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fixes: #6084 Propagate parameter changes to child nodes
```
Comment (svenklemm) [inline .unreleased/bugfix_6084:1]:
```suggestion
Fixes: #6084 Propagate parameter changes to decompress child nodes
```

Comment (jnidzwetzki) [reply .unreleased/bugfix_6084]: Changed

**Review by fabriziomello — APPROVED** [review]

**Pull request merged**